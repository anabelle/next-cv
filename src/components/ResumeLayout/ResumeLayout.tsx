import {
  faBriefcase,
  faCheck,
  faGraduationCap,
} from '@fortawesome/free-solid-svg-icons';
import indefinite from 'indefinite';
import React from 'react';
import { getFullName } from '../../helpers/utils';
import { ResumePageProps } from '../../pages';
import Box from '../../strum-design-system/components/Box/Box';
import Column from '../../strum-design-system/components/Layout/Column';
import Row from '../../strum-design-system/components/Layout/Row';
import Heading from '../../strum-design-system/components/Heading/Heading';
import AboutMe from '../Articles/AboutMe';
import ContactInformation from '../Articles/ContactInformation';
import NotableProjects from '../Articles/NotableProjects';
import EducationItem from '../EducationItem/EducationItem';
import Header from '../Header/Header';
import PageHead from '../PageHead';
import ProfessionalItem from '../ProfessionalItem/ProfessionalItem';
import Section from '../Section/Section';
import SectionHeader from '../SectionHeader/SectionHeader';
import Skills from '../Skills/Skills';
import { PersonalExperiments } from '../PersonalExperiments';
import dynamic from 'next/dynamic';

// Define Footer dynamically
const Footer = dynamic(() => import('../Footer/Footer'), { ssr: false });

const ResumeLayout: React.FC<ResumePageProps> = (props) => {
  const {
    education,
    notableProjects,
    links,
    personalInformation,
    privateInformation,
    professional,
    secret,
    skills,
  } = props;
  const fullName = getFullName(personalInformation);
  const jobTitle = indefinite(personalInformation.attributes.title);

  // Filter skills
  const managementSkill = skills.find(
    (skill) => skill.attributes.title === 'Management & Leadership',
  );
  const languagesSkill = skills.find(
    (skill) => skill.attributes.title === 'Languages',
  );
  const otherSkills = skills.filter(
    (skill) =>
      skill.attributes.title !== 'Management & Leadership' &&
      skill.attributes.title !== 'Languages',
  );

  return (
    <>
      <PageHead
        description={`Professional résumé for ${fullName}, ${jobTitle} living in ${personalInformation.attributes.location}.`}
        personalInformation={personalInformation}
        title={`CV | ${fullName} | ${personalInformation.attributes.location}`}
      />

      <Header secret={secret} {...props} />

      <Section color="standard">
        <Row>
          {/* Main Content Column - Wider */}
          <Column width={{ xs: 12, lg: 8 }}>
            <AboutMe personalInformation={personalInformation} />
          </Column>
          {/* Sidebar Column - Narrower */}
          <Column width={{ xs: 12, lg: 4 }}>
            {/* Render Personal Projects FIRST */}
            <PersonalExperiments />

            <ContactInformation
              personalInformation={personalInformation}
              privateInformation={privateInformation}
            />
            {/* Render Languages skill */}
            {languagesSkill && (
              <Box marginTop={4}>
                <Heading level={4}>{languagesSkill.attributes.title}</Heading>
                <div
                  dangerouslySetInnerHTML={{ __html: languagesSkill.html }}
                />
              </Box>
            )}
          </Column>
        </Row>

        {/* Render remaining skills in the main grid (now below the first row) */}
        {otherSkills.length > 0 && <Skills skills={otherSkills} />}
      </Section>

      <Section color="alternate">
        <Box marginBottom={6}>
          <SectionHeader icon={faBriefcase} text="Professional Experience" />
        </Box>

        {professional.map((experience) => (
          <ProfessionalItem key={experience.slug} {...experience} />
        ))}
      </Section>

      <Section color="standard">
        <Box marginBottom={6}>
          <SectionHeader icon={faGraduationCap} text="Education" />
        </Box>

        {education.map((experience) => (
          <EducationItem key={experience.slug} {...experience} />
        ))}
      </Section>

      <Section color="alternate">
        <NotableProjects notableProjects={notableProjects} />
      </Section>

      <Footer links={links} personalInformation={personalInformation} />
    </>
  );
};

export default ResumeLayout;
