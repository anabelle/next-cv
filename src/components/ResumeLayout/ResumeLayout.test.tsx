import React from 'react';
import {
  educationalExperiences,
  notableProjects,
  links,
  personalInformation,
  professionalExperiences,
  skills,
} from '../../test/fixtures/markdown';
import { render } from '../../test/testUtils';
import ResumeLayout from './ResumeLayout';

jest.mock('next/router', () => require('next-router-mock'));

describe('<ResumeLayout />', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it('Matches the snapshot', () => {
    const { container } = render(
      <ResumeLayout
        education={educationalExperiences}
        notableProjects={notableProjects}
        links={links}
        personalInformation={personalInformation}
        professional={professionalExperiences}
        skills={skills}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
