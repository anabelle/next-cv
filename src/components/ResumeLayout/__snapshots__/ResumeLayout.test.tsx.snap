// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResumeLayout /> Matches the snapshot 1`] = `
<div>
  <header
    class="mock-atoms-class"
  >
    <div
      class="mock-atoms-class Container_containerStyle__rjkhb01 Container_containerFluidStyle__rjkhb00 margin_mxAuto__1nlkou20 mock-atoms-class"
    >
      <div
        class="mock-atoms-class mock-row-style-class"
      >
        <div
          class="mock-atoms-class mock-column-style-base"
        >
          <h1
            class="mock-atoms-class Heading_headingStyleVariants_1__nbnubj1 Heading_headingBaseStyle__nbnubj0"
          >
            Some Body
          </h1>
          <h2
            class="mock-atoms-class Heading_headingStyleVariants_2__nbnubj2 Heading_headingBaseStyle__nbnubj0"
          >
            Cool Person
          </h2>
        </div>
        <div
          class="mock-atoms-class mock-column-style-base mock-col-xs-12 mock-col-sm-12 mock-col-md-auto"
        >
          <a
            class="mock-button-style-class"
            href="/api/pdf"
            target="_blank"
          >
            <i
              class="fa file-pdf"
              data-testid="mock-fa-icon"
            />
            View or Download PDF
          </a>
        </div>
      </div>
    </div>
  </header>
  <section
    class="mock-section-standard"
  >
    <div
      class="mock-atoms-class Container_containerStyle__rjkhb01 Container_containerFluidStyle__rjkhb00 margin_mxAuto__1nlkou20 mock-atoms-class"
    >
      <div
        class="mock-atoms-class mock-row-style-class"
      >
        <div
          class="mock-atoms-class mock-column-style-base mock-col-xs-12"
        >
          <article
            class="mock-atoms-class AboutMe_aboutMeWrapperStyle__1qrrjim0"
          />
          <article
            class="mock-atoms-class"
          />
        </div>
        <div
          class="mock-atoms-class mock-column-style-base mock-col-xs-12 mock-col-lg-4"
        >
          <section
            class="mock-atoms-class"
          >
            <h2
              class="PersonalExperiments_sectionTitleStyle__16fyf952"
            >
              <i
                class="fa code"
                data-testid="mock-fa-icon"
              />
              <span>
                Personal Projects
              </span>
            </h2>
            <div
              class="PersonalExperiments_projectContainerStyle__16fyf95h"
            >
              <a
                aria-label="Pixel (opens in new tab)"
                class="PersonalExperiments_projectCardFeatured__16fyf955 mock-atoms-class"
                href="https://pixel.xx.kg"
                rel="noopener noreferrer"
                style="text-decoration: none;"
                target="_blank"
              >
                <div
                  class="PersonalExperiments_imageBannerFeaturedStyle__16fyf957 mock-atoms-class"
                >
                  <img
                    alt="Pixel screenshot"
                    class=""
                    data-nimg="fill"
                    decoding="async"
                    loading="lazy"
                    sizes="100vw"
                    src="/_next/image?url=%2Fimages%2Fpixel.png&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fpixel.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fpixel.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fpixel.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fpixel.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fpixel.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fpixel.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fpixel.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fpixel.png&w=3840&q=75 3840w"
                    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                  />
                  <div
                    class="PersonalExperiments_featuredAvatarWrapper__16fyf958"
                  >
                    <img
                      alt="Pixel avatar"
                      class="PersonalExperiments_featuredAvatarImg__16fyf959"
                      src="/images/pixel_avatar.png"
                    />
                  </div>
                </div>
                <div
                  class="PersonalExperiments_contentAreaStyle__16fyf95b mock-atoms-class"
                >
                  <span
                    class="PersonalExperiments_imageTitleStyle__16fyf95a mock-atoms-class"
                  >
                    Pixel
                  </span>
                  <div
                    class="PersonalExperiments_techListStyle__16fyf95c mock-atoms-class"
                  >
                    <span
                      class="PersonalExperiments_techTagStyle__16fyf95d mock-atoms-class"
                    >
                      Autonomous AI Agent
                    </span>
                    <span
                      class="PersonalExperiments_techTagStyle__16fyf95d mock-atoms-class"
                    >
                      Digital Art
                    </span>
                    <span
                      class="PersonalExperiments_techTagStyle__16fyf95d mock-atoms-class"
                    >
                      Bitcoin
                    </span>
                    <span
                      class="PersonalExperiments_techTagStyle__16fyf95d mock-atoms-class"
                    >
                      Nostr
                    </span>
                  </div>
                  <p
                    class="PersonalExperiments_descriptionStyle__16fyf95e mock-atoms-class"
                  >
                    Survival‑driven pixel art agent on a $3/mo VPS: sells pixels for sats (1 sat = 1 pixel) on a live collaborative canvas. Built Lightning payments, real‑time rendering, Telegram bot, and Nostr integrations.
                  </p>
                  <span
                    class="PersonalExperiments_viewLinkStyle__16fyf95f mock-atoms-class"
                  >
                    Meet Pixel
                    <i
                      class="fa up-right-from-square"
                      data-testid="mock-fa-icon"
                    />
                  </span>
                </div>
              </a>
              <a
                aria-label="TetrisTwist (opens in new tab)"
                class="PersonalExperiments_projectCardNonLink__16fyf954 mock-atoms-class"
                href="https://tetristwist.heyanabelle.com"
                rel="noopener noreferrer"
                style="text-decoration: none;"
                target="_blank"
              >
                <div
                  class="PersonalExperiments_imageBannerStyle__16fyf956 mock-atoms-class"
                >
                  <img
                    alt="TetrisTwist screenshot"
                    class=""
                    data-nimg="fill"
                    decoding="async"
                    loading="lazy"
                    sizes="(max-width: 768px) 100vw, 50vw"
                    src="/_next/image?url=%2Fimages%2Ftetris.png&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Ftetris.png&w=384&q=75 384w, /_next/image?url=%2Fimages%2Ftetris.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2Ftetris.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2Ftetris.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2Ftetris.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Ftetris.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Ftetris.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Ftetris.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Ftetris.png&w=3840&q=75 3840w"
                    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                  />
                </div>
                <div
                  class="PersonalExperiments_contentAreaStyle__16fyf95b mock-atoms-class"
                >
                  <span
                    class="PersonalExperiments_imageTitleStyle__16fyf95a mock-atoms-class"
                  >
                    TetrisTwist
                  </span>
                  <div
                    class="PersonalExperiments_techListStyle__16fyf95c mock-atoms-class"
                  >
                    <span
                      class="PersonalExperiments_techTagStyle__16fyf95d mock-atoms-class"
                    >
                      Three.js
                    </span>
                    <span
                      class="PersonalExperiments_techTagStyle__16fyf95d mock-atoms-class"
                    >
                      WebGL
                    </span>
                    <span
                      class="PersonalExperiments_techTagStyle__16fyf95d mock-atoms-class"
                    >
                      React
                    </span>
                    <span
                      class="PersonalExperiments_techTagStyle__16fyf95d mock-atoms-class"
                    >
                      TypeScript
                    </span>
                  </div>
                  <p
                    class="PersonalExperiments_descriptionStyle__16fyf95e mock-atoms-class"
                  >
                    3D Tetris‑style game in Three.js/WebGL with a lightweight physics loop, smooth transforms, and responsive controls.
                  </p>
                  <span
                    class="PersonalExperiments_viewLinkStyle__16fyf95f mock-atoms-class"
                  >
                    View Project
                    <i
                      class="fa up-right-from-square"
                      data-testid="mock-fa-icon"
                    />
                  </span>
                </div>
              </a>
              <a
                aria-label="Multiplayer Snake (opens in new tab)"
                class="PersonalExperiments_projectCardNonLink__16fyf954 mock-atoms-class"
                href="https://snake.heyanabelle.com"
                rel="noopener noreferrer"
                style="text-decoration: none;"
                target="_blank"
              >
                <div
                  class="PersonalExperiments_imageBannerStyle__16fyf956 mock-atoms-class"
                >
                  <img
                    alt="Multiplayer Snake screenshot"
                    class=""
                    data-nimg="fill"
                    decoding="async"
                    loading="lazy"
                    sizes="(max-width: 768px) 100vw, 50vw"
                    src="/_next/image?url=%2Fimages%2Fsnake.png&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fsnake.png&w=384&q=75 384w, /_next/image?url=%2Fimages%2Fsnake.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fsnake.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fsnake.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fsnake.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fsnake.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fsnake.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fsnake.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fsnake.png&w=3840&q=75 3840w"
                    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                  />
                </div>
                <div
                  class="PersonalExperiments_contentAreaStyle__16fyf95b mock-atoms-class"
                >
                  <span
                    class="PersonalExperiments_imageTitleStyle__16fyf95a mock-atoms-class"
                  >
                    Multiplayer Snake
                  </span>
                  <div
                    class="PersonalExperiments_techListStyle__16fyf95c mock-atoms-class"
                  >
                    <span
                      class="PersonalExperiments_techTagStyle__16fyf95d mock-atoms-class"
                    >
                      Socket.IO
                    </span>
                    <span
                      class="PersonalExperiments_techTagStyle__16fyf95d mock-atoms-class"
                    >
                      Canvas API
                    </span>
                    <span
                      class="PersonalExperiments_techTagStyle__16fyf95d mock-atoms-class"
                    >
                      Node.js
                    </span>
                    <span
                      class="PersonalExperiments_techTagStyle__16fyf95d mock-atoms-class"
                    >
                      React
                    </span>
                  </div>
                  <p
                    class="PersonalExperiments_descriptionStyle__16fyf95e mock-atoms-class"
                  >
                    Real‑time multiplayer Snake with client‑server sync, collision handling, and persistent leaderboards.
                  </p>
                  <span
                    class="PersonalExperiments_viewLinkStyle__16fyf95f mock-atoms-class"
                  >
                    View Project
                    <i
                      class="fa up-right-from-square"
                      data-testid="mock-fa-icon"
                    />
                  </span>
                </div>
              </a>
            </div>
          </section>
        </div>
      </div>
      <div
        class="mock-atoms-class"
      />
    </div>
  </section>
  <section
    class="mock-section-alternate"
  >
    <div
      class="mock-atoms-class Container_containerStyle__rjkhb01 Container_containerFluidStyle__rjkhb00 margin_mxAuto__1nlkou20 mock-atoms-class"
    >
      <div
        class="mock-atoms-class"
      />
      <article
        class="article_articleStyle__on6lov0"
      >
        <h4
          class="mock-atoms-class Heading_headingStyleVariants_4__nbnubj4 Heading_headingBaseStyle__nbnubj0"
        >
          <span
            class="mock-atoms-class"
          >
            My Title
          </span>
           
          at 
          The Zebra
        </h4>
        <div>
          <i
            class="fa calendar"
            data-testid="mock-fa-icon"
          />
          2018-02-15
          –
          Current
        </div>
        <div
          class="mock-atoms-class"
        >
          <div>
            Test
          </div>
        </div>
      </article>
    </div>
  </section>
  <section
    class="mock-section-standard"
  >
    <div
      class="mock-atoms-class Container_containerStyle__rjkhb01 Container_containerFluidStyle__rjkhb00 margin_mxAuto__1nlkou20 mock-atoms-class"
    >
      <div
        class="mock-atoms-class"
      />
      <article
        class="article_articleStyle__on6lov0"
      >
        <h4
          class="mock-atoms-class Heading_headingStyleVariants_4__nbnubj4 Heading_headingBaseStyle__nbnubj0"
        >
          Degree
        </h4>
        <div>
          <i
            class="fa building-columns"
            data-testid="mock-fa-icon"
          />
          Test School
        </div>
        <div>
          <div>
            Test
          </div>
        </div>
      </article>
    </div>
  </section>
  <section
    class="mock-section-alternate"
  >
    <div
      class="mock-atoms-class Container_containerStyle__rjkhb01 Container_containerFluidStyle__rjkhb00 margin_mxAuto__1nlkou20 mock-atoms-class"
    >
      <article>
        <h3
          class="mock-atoms-class Heading_headingStyleVariants_3__nbnubj3 Heading_headingBaseStyle__nbnubj0"
        >
          <span
            class="mock-atoms-class fa-layers fa-fw"
          >
            <i
              class="fa circle"
              data-testid="mock-fa-icon"
            />
            <i
              class="fa trophy"
              data-testid="mock-fa-icon"
            />
          </span>
          Awards & Featured Projects
        </h3>
        <div>
          <div>
            Test
          </div>
        </div>
      </article>
    </div>
  </section>
</div>
`;
